### PR TITLE
Clarify how insertOrUpdate behave for objects with primary keys.

### LINKED ISSUE #3109: Fix insertOrUpdate Javadoc
Feedback: 
> from the javadoc of `insertOrUpdate()` We don't check if the provided objects are already managed or not, so inserting a managed object might duplicate it` (https://github.com/realm/realm-java/blob/master/realm/realm-library/src/main/java/io/realm/Realm.java#L863) this seems confusing in my opinion, it lead me to think that `insertOrUpdate()` might duplicate objects.

It is true that `insertOrUpdate` never duplicate objects with primary keys. We should clarify that.

### PR DESCRIPTION
**Pull request by cmelchior**
Fixes #3109 

@realm/java

### CODE REVIEW

**Comment by beeender** [comment]
:+1:

**Comment by stk1m1** [comment]
👍

File: realm/realm-library/src/main/java/io/realm/Realm.java [hunk 1]
```diff
@@ -840,7 +840,10 @@ public void insert(RealmModel object) {
      * <p>
      * Please note:
      * <ul>
-     * <li>We don't check if the provided objects are already managed or not, so inserting a managed object might duplicate it</li>
+     * <li>
+     *     We don't check if the provided objects are already managed or not, so inserting a managed object might duplicate it.
+     *     This will only happen if the object doesn't have a primary key. Objects with primary keys will never get duplicated.
```
Comment (kneth) [inline realm/realm-library/src/main/java/io/realm/Realm.java]: "This" -> "Duplication"

File: realm/realm-library/src/main/java/io/realm/Realm.java [hunk 2]
```diff
@@ -871,7 +874,10 @@ public void insertOrUpdate(Collection<? extends RealmModel> objects) {
...
```
Comment (kneth) [inline realm/realm-library/src/main/java/io/realm/Realm.java]: "This" -> "Duplication"

**Comment by kneth** [comment]
:+1: with minor comment

File: realm/realm-library/src/main/java/io/realm/Realm.java (hunk 1 above)
Comment (zaki50) [inline realm/realm-library/src/main/java/io/realm/Realm.java]: Shouldn't we add [at]throws clause in this javadoc comment as well?

Comment (cmelchior) [inline realm/realm-library/src/main/java/io/realm/Realm.java]: Done

**Pull request merged**